src/index.js

```diff
@@ -26,115 +25,149 @@ const WARN = 'warning';
 const ERROR = 'error';
 
 const getBaseComponent = (components, component) => {
-  if (component in components) {
-    return components[component];
-  } else throw new Error(`No component defined with name ${component}`)
+	if (component in components) {
+		return components[component];
+	} else {
+		const error = new Error(`No component defined with name ${component}`);
+		error.internal = true;
+		throw error;
```

Comment:
I think I'd favor a custom error subclass if the intent of this is to later recognize internal errors. Something like

```javascript
export default class InternalError extends Error {};
```

```javascript
throw new InternalError("...");
```

```javascript
catch (error) {
  if (error instanceof InternalError) { ... }
}
```